--- a/src/gscam_node.cpp
+++ b/src/gscam_node.cpp
@@ -362,7 +362,6 @@ void GSCamNode::impl::process_frame()
   // Stop on end of stream
   if (!buf) {
-    RCLCPP_INFO(node_->get_logger(), "Stream ended, pause for 1s");
-    using namespace std::chrono_literals;
-    std::this_thread::sleep_for(1s);
-    return;
+    RCLCPP_ERROR(node_->get_logger(), "Stream ended, exiting");
+    delete_pipeline();
+    exit(1);
   }
